craft_providers/lxd/lxd_provider.py

```diff
@@ -87,6 +88,15 @@ def is_provider_installed(cls) -> bool:
         """
         return is_installed()
 
+    def list_instances(self) -> Collection[LXDInstance]:
```

Comment:
suggestion: Add a test, at least to make sure we detect that this function should be updated if/when the LXDInstance class is updated? (same for the multipass provider)